### DISCUSSION TITLE
Migrating from Native to Capacitor (PWA)

### DISCUSSION THREAD

**Discussion by CTOlet** [ionic-team/capacitor#4728] [Q&A]
Hello everybody!

We have (1) an Android native app and (2) React Native iOS app using Firebase tokens for push notifications. 
Can we migrate the both of them to Ionic Capacitor and save an old Firebase tokens, because we have a users storage on the backend that identified by Firebase token?

**Comment by jcesarmobile** [comment]
Push tokens can change at any time, so shouldn't be used as a way of identify users. 
What's going to be harder is to maintain the user session, they will probably need to login again, and in that moment you can associate their user and token again in your servers (that could be the same push token or a new one)

**Reply by CTOlet** [reply]
Yes, in the other words, I am looking for solution to not force users to login again after update (replacing native app with capacitor app).

Do anyone have an idea how we can solve that?

**Reply by jcesarmobile** [reply]
Will depend on how are you saving the session at the moment, but will probably not be possible or will require a lot of work.
